### PR TITLE
[CR] Unicode support for SCT

### PR DESCRIPTION
**Pull request by DarkDaskin**
Scrolling combat text is currently rendered using ASCII tiles. This produces garbage for all non-ASCII characters.

This PR adds a new SCT rendering mode which uses configured `map_font` to draw on top of tiles. I have added the `ANIMATION_SCT_USE_FONT` setting (enabled by default) to switch between old and new rendering modes.

Terrain overlay text rendering code is decoupled from SCT and can be reused to draw text during other stages of terrain rendering.

Not sure if the current font is readable enough, or we should use bigger font by default. ASCII tiles are much bigger. Also we might want to add a background or outline to overlay text for better readability.

The code may contain some stupid mistakes because I'm new to the project and rarely write C++, please review. I didn't find any problems during a quick playtest.

### CODE REVIEW

**Comment by AMurkin** [comment]
You shouldn't edit pot and po files directly. See https://github.com/CleverRaven/Cataclysm-DDA/blob/master/.github/CONTRIBUTING.md#translations

**Comment by DarkDaskin** [comment]
Tried to push my changes to Transifex and got the following error: `tx ERROR: Error received from server: Only maintainers are allowed to update the source file.`

Should I revert the translation files and wait until the PR is merged and a maintainer updates the Transifex project?

**Comment by BrettDong** [comment]
Just editing C++ source files is fine. After this PR gets merged, I'll add them to our translation system in the next routine i18n update cycle.

**Comment by Leland** [comment]
Need to check how this looks on a mac

**Comment by AMurkin** [comment]
@Leland From you screenshots - not the text only, but tiles too.

File: src/sdltiles.cpp [hunk 1]
```diff
@@ -851,14 +851,67 @@ void cata_cursesport::curses_drawwindow( const catacurses::window &w )
...
+            }    
+
+            int alignment_offset = 0;
+            if ( ft.alignment == TEXT_ALIGNMENT_CENTER ) {
+                alignment_offset = full_text_length / 2;
+            } else if ( ft.alignment == TEXT_ALIGNMENT_RIGHT ) {
+                alignment_offset = full_text_length - 1;
+            }
+
+            // TODO: ensure it works with composite chars
```
Comment (AMurkin) [inline src/sdltiles.cpp]: Is this still in question?

Comment (DarkDaskin) [reply src/sdltiles.cpp]: Checked how it works, my code passes correct substrings to `OutputChar`, but it still renders incorrectly. It does render the same way in other parts of the game though.

**Comment by DarkDaskin** [comment]
@Leland I had the same problem with a recent build from master branch on Windows. Probably you triggered it simply by going into options. It only reproduces for me when the window is maximized.

**Comment by Leland** [comment]
Oh, interesting! Will check soon

**Comment by Leland** [comment]
Can confirm the stretching issue happens on master as well

**Comment by AMurkin** [comment]
So, except for the merge conflict (take changes from this branch, but set 0 value to `screen_x` and `screen_y` in `player_to_screen` to resolve), it's good.

**Comment by ZhilkinSerg** [comment]
We should probably add SCT Unicode font size option - SCT is small and barely visible now:

image

**Comment by DarkDaskin** [comment]
@ZhilkinSerg You can fiddle with the settings in `config/fonts.json`. Try to increase the `map_*` values proportionally, e.g.

```
"map_fontwidth": 16,
"map_fontheight": 32,
"map_fontsize": 32,
```
The default font will look pixelated at larger sizes, but this will fit your tileset nicely.

For a more general case I think that we should implement font settings UI for all parts of the game (in a separate PR) rather than just for SCT. Just changing the font size without replacing the font will not look good in most cases. Although we can add a simple integer font scale factor setting for SCT as a temporary measure.

**Comment by ZhilkinSerg** [comment]
> For a more general case I think that we should implement font settings UI for all parts of the game (in a separate PR) rather than just for SCT.

I agree.

**Pull request merged**